kbs/src/api/src/attestation/mod.rs

```diff
@@ -32,6 +34,21 @@ pub trait Attest: Send + Sync {
     /// Verify Attestation Evidence
     /// Return Attestation Results Token
     async fn verify(&self, tee: Tee, nonce: &str, attestation: &str) -> Result<String>;
+
+    /// generate the Challenge to pass to attester based on Tee and nonce
+    async fn generate_challenge(&self, _tee: Tee, _tee_parameters: String) -> Result<Challenge> {
+        let mut nonce: Vec<u8> = vec![0; 32];
+
+        thread_rng()
+            .try_fill(&mut nonce[..])
+            .map_err(anyhow::Error::from)?;
+
+        let nonce = STANDARD.encode(&nonce);
+        Ok(Challenge {
+            nonce,
+            extra_params: String::new(),
+        })
```

Comment:
Do we really need to implement this method here? It's just a fallback, right?

Reply:
Yes, this is a fallback.